src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -0,0 +1,294 @@
+                            colony.getMessageEntityPlayers(),
+                            "tile.blockHutTownHall.messageMaxSize",
+                            colony.getName());
+                }
+            }
+            entity.setColony(colony, citizenData);
+
+            entity.setPosition(spawnPoint.getX() + HALF_BLOCK, spawnPoint.getY() + SLIGHTLY_UP, spawnPoint.getZ() + HALF_BLOCK);
+            world.spawnEntity(entity);
+
+            colony.checkAchievements();
+            markCitizensDirty();
+        }
+    }
+
+    @Override
+    public void removeCitizen(@NotNull final CitizenData citizen, @NotNull final Colony colony)
+    {
+        //Remove the Citizen
+        citizens.remove(citizen.getId());
+
+        if (citizen.getWorkBuilding() != null)
+        {
+            citizen.getWorkBuilding().cancelAllRequestsOfCitizen(citizen);
+        }
+
+        if (citizen.getHomeBuilding() != null)
+        {
+            citizen.getHomeBuilding().cancelAllRequestsOfCitizen(citizen);
+        }
+
+        for (@NotNull final AbstractBuilding building : colony.getBuildingManager().getBuildings().values())
+        {
+            building.removeCitizen(citizen);
+        }
+
+        colony.getWorkManager().clearWorkForCitizen(citizen);
+
+        //  Inform Subscribers of removed citizen
+        for (final EntityPlayerMP player : colony.getSubscribers())
+        {
+            MineColonies.getNetwork().sendTo(new ColonyViewRemoveCitizenMessage(colony, citizen.getId()), player);
+        }
+    }
+
+    @Override
+    public CitizenData getJoblessCitizen()
+    {
+        for (@NotNull final CitizenData citizen : citizens.values())
+        {
+            if (citizen.getWorkBuilding() == null)
+            {
+                return citizen;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public void calculateMaxCitizens(@NotNull final Colony colony)
+    {
+        int newMaxCitizens = 0;
+
+        for (final AbstractBuilding b : colony.getBuildingManager().getBuildings().values())
+        {
+            if (b.getBuildingLevel() > 0)
+            {
+                if (b instanceof BuildingHome)
+                {
+                    newMaxCitizens += ((BuildingHome) b).getMaxInhabitants();
+                }
+                else if (b instanceof BuildingBarracksTower)
+                {
+                    newMaxCitizens += b.getBuildingLevel();
+                }
+            }
+        }
+        // Have at least the minimum amount of citizens
+        newMaxCitizens = Math.max(Configurations.gameplay.maxCitizens, newMaxCitizens);
+        if (getMaxCitizens() != newMaxCitizens)
+        {
+            setMaxCitizens(newMaxCitizens);
+            colony.markDirty();
+        }
+    }
+
+    /**
+     * Spawn a brand new Citizen.
+     */
+    public void spawnCitizen(@NotNull final Colony colony)
+    {
+        spawnCitizen(null, colony.getWorld(), colony.getBuildingManager(), colony);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Parameter 1 to this call is marked "org.jetbrains.annotations.NotNull" but null is passed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2637)